### PR TITLE
Update release process documentation wrt bintray removal and new usage of ASF JFrog Artifactory

### PR DESCRIPTION
**Pull request by michaelsembwever**

### CODE REVIEW

**Comment by michaelsembwever** [comment]
@ekaterinadimitrova2 up for a review?

File: doc/source/development/release_process.rst [hunk 1]
```diff
@@ -159,10 +159,6 @@ Promote Nexus Repository
 * Select the "Releases" repository and click "Promote".
 * Next click on "Repositories", select the "Releases" repository and validate that your artifacts exist as you expect them.
 
-Publish the Bintray Uploaded Distribution Packages
----------------------------------------
-
-Log into bintray and publish the uploaded artifacts.
```
Comment (ekaterinadimitrova2) [inline doc/source/development/release_process.rst:165]: I am not familiar with this artifactory but I assume we will need similar step here?

**Review by ekaterinadimitrova2 — APPROVED** [review]
Left one comment I wasn't sure about, otherwise looks good to me

File: doc/source/development/release_process.rst (hunk 1 above)
Comment (michaelsembwever) [reply doc/source/development/release_process.rst:165]: No. It has not such "publish" step. What's uploaded to it is immediately available to download.

Comment (ekaterinadimitrova2) [reply doc/source/development/release_process.rst:165]: Great, thanks for confirming :-)

**Comment by michaelsembwever** [comment]
merged manually with https://github.com/apache/cassandra/commit/ab91b1fdc7ce253fa00a8b68d45e4710b8819bd8

**Pull request closed without merging**